apps/prairielearn/src/pages/instructorAssessmentAccessOverrides/instructorAssessmentAccessOverrides.js

```diff
@@ -0,0 +1,132 @@
+// @ts-check
+import * as express from 'express';
+const asyncHandler = require('express-async-handler');
+import * as sqldb from '@prairielearn/postgres';
+import * as error from '@prairielearn/error';
+import { getEnrollmentForUserInCourseInstance } from '../../models/enrollment';
+import { selectUserByUid } from '../../models/user';
+
+const router = express.Router();
+const sql = sqldb.loadSqlEquiv(__filename);
+
+router.get(
+  '/',
+  asyncHandler(async (req, res) => {
+    const result = await sqldb.queryAsync(sql.select_assessment_access_policies, {
+      assessment_id: res.locals.assessment.id,
+      student_uid: req.body.student_uid,
+    });
+    res.locals.policies = result.rows;
+    // console.log(res.locals);
+    res.render(__filename.replace(/\.js$/, '.ejs'), res.locals);
+  }),
+);
+
+router.post(
+  '/',
+  asyncHandler(async (req, res) => {
+    if (req.body.__action === 'add_new_override') {
+      const params = {
+        assessment_id: res.locals.assessment.id,
+        created_by: res.locals.authn_user.user_id,
+        credit: req.body.credit,
+        end_date: new Date(req.body.end_date),
+        group_name: req.body.group_name || null,
+        note: req.body.note || null,
+        start_date: new Date(req.body.start_date),
+        user_id: req.body.student_uid,
```

Comment:
I'd strongly advise against storing a `uid` in a property named `user_id`; this is really confusing to read later on.

Reply:
Question: 52ad2e9197df6e4f0d1f3b03edb191954f55a44f has the correction for passing the user id but I believe I am doing it wrong because I am searching all students and not just those enrolled, additionally, I am searching with student_uid even if it is a group type assessment. The reason I feel like I need to get the user before I define the params is because I used const to define the params and dont know how to initalize it without setting it to null.

Reply:
Done in 52ad2e9

Reply:
^ Comment removed ac713bfaa4f2c8a3b9ae9de7b6776554fece4e51